src/ophyd_async/fastcs/panda/_hdf_panda.py

```diff
@@ -13,17 +13,15 @@
 class HDFPanda(CommonPandaBlocks, StandardDetector):
     def __init__(
         self,
-        prefix: str,
+        uri: str,
```

Comment:
Is it possible to please retain prefix so that I don't have to modify `device_instantiation()` in dodal to explicitly not send it?

> Robert Tuck
  Today at 10:17 AM
Thanks. After some further investigation, it looks like it will lead to some ugliness in dodal's device_instantiation because the panda doesn't accept a prefix parameter, whereas every single other dodal device does, so I have to add code to explicitly not send prefix. Can it be made an optional keyword parameter with backwards compatibility?
New


> Tom Cobb
>   [4 minutes ago](https://diamondlightsource.slack.com/archives/C04RRAM68LD/p1729681756673319?thread_ts=1729600610.523609&cid=C04RRAM68LD)
> I can put it back to prefix , and change it back to uri when device_instantiation has gone and device_factory is used instead

Reply:
FWIW we're ultimately going to replace `device_instantiation` with `DeviceInitialisationFactory`.